app/controllers/fact_values_controller.rb

```diff
@@ -24,12 +25,30 @@ def index
       values = values.root_only
     end
 
-    @fact_values = values.no_timestamp_facts.required_fields.paginate :page => params[:page]
+    @fact_values = values.no_timestamp_facts
+
+    respond_to do |format|
+      format.html do
+        @fact_values = @fact_values.preload(related_tables).paginate :page => params[:page]
```

Comment:
for a small size (20 rows by default) preloading the related tables is faster then joining.